### PR TITLE
Feat/abdm user abha creation

### PR DESCRIPTION
**Pull request by yashasva01**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
+require('dotenv').config()
+const express = require('express')
+const sequelize = require('./src/services/database/dbConnection')
```
Comment (rkdixit96) [inline index.js:3]: Not needed here

File: src/fetchedToken.txt [hunk 2]
```diff
@@ -0,0 +1 @@
...nb3YuaW4vYXV0aC9yZWFsbXMvY2VudHJhbC1yZWdpc3RyeSIsImF1ZCI6ImFjY291bnQiLCJzdWIiOiI0NDM2M2JkYy1mNjNmLTQyN2YtOGY0OS1iY2JkNzhiNDU4MWMiLCJ0eXAiOiJCZWFyZXIiLCJhenAiOiJTQlhfMDAyMTQ4Iiwic2Vzc2lvbl9zdGF0ZSI6ImU1MDYyN2IwLWUxMzUtNDViNC05YmIxLWEyNjQzYjU3ZmQ1OCIsImFjciI6IjEiLCJhbGxvd2VkLW9yaWdpbnMiOlsiaHR0cDovL2xvY2FsaG9zdDo5MDA3Il0sInJlYWxtX2FjY2VzcyI6eyJyb2xlcyI6WyJoaXUiLCJvZmZsaW5lX2FjY2VzcyIsImhlYWx0aElkIiwiT0lEQyIsImhpcCJdfSwicmVzb3VyY2VfYWNjZXNzIjp7IlNCWF8wMDIxNDgiOnsicm9sZXMiOlsidW1hX3Byb3RlY3Rpb24iXX0sImFjY291bnQiOnsicm9sZXMiOlsibWFuYWdlLWFjY291bnQiLCJtYW5hZ2UtYWNjb3VudC1saW5rcyIsInZpZXctcHJvZmlsZSJdfX0sInNjb3BlIjoib3BlbmlkIGVtYWlsIHByb2ZpbGUiLCJjbGllbnRJZCI6IlNCWF8wMDIxNDgiLCJjbGllbnRIb3N0IjoiMTAuMjMzLjY3LjEzMiIsImVtYWlsX3ZlcmlmaWVkIjpmYWxzZSwicHJlZmVycmVkX3VzZXJuYW1lIjoic2VydmljZS1hY2NvdW50LXNieF8wMDIxNDgiLCJjbGllbnRBZGRyZXNzIjoiMTAuMjMzLjY3LjEzMiJ9.QnwMddOkRXHDCN05Vnw6y8v94eSzRaBUWeUh5A5JdMiw3hPBHYvxlBrRVxH39lXuwiFFC7-l92sHmWs5RrSX1PuyDi1luLvzrCCTThoQcxxOnMmx1JQI1VvIGYVhuYfH2DP7_2p7LyyinOWS48rXlAClYizGaPAgybc_aKhgt1is-vFfkYuNeZ-xK0817BaqqLVu5Rrt3dDwNmUd2VlKlqUJsB8GC4Z75nydSEF4dt_ZSNgJG9JhRm0z8y0Re0Ghzz1woNMDk9SDV2JrkCOwpQVt4R5jHbKHPvDj3RGMZzB0xg9LHmqEpXINg0BMqm1iv_H3u_UZt0bIQ3SZd6RbkQ
```
[the remark this answers is not in the document]
Comment (rkdixit96) [reply src/fetchedToken.txt:1]: Do a git rm

File: src/routes/onboarding.js [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+router.post(
+  "/users/checkAndGenerateMobileOTP",
+  user.handleCheckAndGenerateMobileOTP
+);
+
+router.post('/users/verifyMobileOTP', user.verifyMobileOTP);
+
+router.post('/users/createHeathIDPreVerifiedNumber', user.createHeathIDPreVerifiedNumber);
+
+//router.post('users/', user.resendOTP)
```
Comment (rkdixit96) [inline src/routes/onboarding.js:23]: Remove commented route

**Review by rkdixit96 — CHANGES_REQUESTED** [review]

**Comment by rkdixit96** [comment]
Remove all commented code @yashasva01

**Comment by yashasva01** [comment]
> Remove all commented code @yashasva01

okay, will edit.

File: index.js (hunk 1 above)
Comment (yashasva01) [reply index.js:3]: Done

File: index.js [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+const app = express()
+const onboardingRoutes = require('./src/routes/onboarding')
+
+// sequelize.authenticate()
+//   .then(() => console.log('Connection to Database was successful'))
+//   .catch(console.log)
```
[the remark this answers is not in the document]
Comment (yashasva01) [reply index.js:9]: Done

File: src/util/abdm.js [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+const fs = require('fs')
+const FILE_NAME = 'fetchedToken.txt'
+const tokenFileExists = () =>
+  fs.existsSync(FILE_NAME)
+
+const parseJWT = (jwt) => jose.decodeJwt(jwt)
+
+const fetchJWTFromServer = async () => {
+  //console.log(process.env.SANDBOXURL)
+  const response = await fetch('https://gjgew5guy33vice6fauegfokme0dyaxy.lambda-url.us-east-1.on.aws/?code=krkedikhayenge')
```
[the remark this answers is not in the document]
Comment (yashasva01) [reply src/util/abdm.js:15]: Done

File: src/util/userOnboardingApi.js [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+    },{
+      headers: {
+          Authorization: `Bearer ${token}`,
+      }
+    });
+    return response;
+}
+
+const createHealthIdWithPreVerified = async (userData, userDetails, txnId, token) => {
+    const response = await axios.post('https://healthidsbx.abdm.gov.in/api/v1/registration/aadhaar/createHealthIdWithPreVerified', {
```
[the remark this answers is not in the document]
Comment (yashasva01) [reply src/util/userOnboardingApi.js:56]: Done

File: src/util/userOnboardingApi.js [hunk 7]
```diff
@@ -0,0 +1,82 @@
+const axios = require('axios');
+const urlEndpoint = require('../config/User/onboardingEndpoints')
+
+
```
Comment (rkdixit96) [inline src/util/userOnboardingApi.js:4]: All of these are services

File: src/config/User/onboardingEndpoints.js [hunk 8]
```diff
@@ -0,0 +1,20 @@
...
+
+
+
+const generateOtpUrl = 'https://healthidsbx.abdm.gov.in/api/v2/registration/aadhaar/generateOtp'
+const verifyOtpUrl = 'https://healthidsbx.abdm.gov.in/api/v2/registration/aadhaar/verifyOTP'
+const resendOtpUrl = 'https://healthidsbx.abdm.gov.in/api/v1/registration/aadhaar/resendAadhaarOtp'
+const checkAndGenerateMobileOtpUrl = 'https://healthidsbx.abdm.gov.in/api/v2/registration/aadhaar/checkAndGenerateMobileOTP'
+const verifyMobilOtpUrl = 'https://healthidsbx.abdm.gov.in/api/v1/registration/aadhaar/verifyMobileOTP'
+const createHealthIdUrl = 'https://healthidsbx.abdm.gov.in/api/v1/registration/aadhaar/createHealthIdWithPreVerified'
+
```
Comment (rkdixit96) [inline src/config/User/onboardingEndpoints.js:11]: Extract out base url - `https://healthidsbx.abdm.gov.in` from all these endpoints
So when we move to PROD we have to make a change in only one place

File: src/util/userOnboardingApi.js (hunk 7 above)
Comment (rkdixit96) [reply src/util/userOnboardingApi.js:4]: Just make these API calls directly in the service

**Review by rkdixit96 — CHANGES_REQUESTED** [review]
Remove .DS_Store file. Add that to .gitignore as well

File: src/config/User/onboardingEndpoints.js (hunk 8 above)
Comment (maitra100) [reply src/config/User/onboardingEndpoints.js:11]: done

File: database/migrations/20230213052619-create-users.js [hunk 9]
```diff
@@ -0,0 +1,61 @@
...
+        type: Sequelize.STRING
+      },
+      gender: {
+        type: Sequelize.STRING
+      },
+      profilePhoto: {
+        type: Sequelize.STRING
+      },
+      emailId: {
+        type: Sequelize.INTEGER
```
Comment (NiranjanBorawar) [inline database/migrations/20230213052619-create-users.js:43]: can we check type of all the fields once again
like emailId should be string, dayOfBirth and yearOfBirth we can keep as Integer, Gender we can try to keep as ENUM

File: database/migrations/20230213053309-create-user-location.js [hunk 10]
```diff
@@ -0,0 +1,44 @@
...
+        type: Sequelize.STRING
+      },
+      districtCode: {
+        type: Sequelize.STRING
+      },
+      subDistrictCode: {
+        type: Sequelize.STRING
+      },
+      pinCode: {
+        type: Sequelize.STRING
```
Comment (NiranjanBorawar) [inline database/migrations/20230213053309-create-user-location.js:29]: pinCode should be INTEGER

File: database/migrations/20230213052619-create-users.js (hunk 9 above)
Comment (maitra100) [reply database/migrations/20230213052619-create-users.js:43]: emailId is a foreign key referencing the primary key of User_email table. dayOfBirth and yearOfBirth have been removed in the next migration files. We'll change the type of gender.

File: database/migrations/20230213053309-create-user-location.js (hunk 10 above)
Comment (maitra100) [reply database/migrations/20230213053309-create-user-location.js:29]: Krishna said let it be string.

File: src/util/userOnboardingApi.js (hunk 7 above)
Comment (maitra100) [reply src/util/userOnboardingApi.js]: done

**Review by rkdixit96 — APPROVED** [review]

**Review by SaatwikRishi — APPROVED** [review]

**Pull request merged**